### PR TITLE
XRandR Display Extension

### PR DESCRIPTION
**Pull request by RobertBColton**
Taking some steps here to address part of #1064 and implement the display functions for xlib using the XRandR extension. I did this because it is pretty much standard and Hugar specifically requested `display_set_size` and he told me he'd be ok if I use that. I followed this guide from the OpenGL wiki to write the code:
https://www.khronos.org/opengl/wiki/Programming_OpenGL_in_Linux:_Changing_the_Screen_Resolution
I followed this Linux documentation as well:
https://linux.die.net/man/3/xrrsetscreenconfig

Depth switching is not provided by XRandR, so `display_set_colordepth` is still not implemented and `display_test_all`/`display_set_all` will ignore that same parameter.

This adds `libxrandr-dev` (`-lXrandr`) as a dependency, but only when the user wants to use this extension.

### CODE REVIEW

**Comment by RobertBColton** [comment]
@JoshDreamland this one is ready to go when you get a chance, thanks!

File: ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey [hunk 1]
```diff
@@ -0,0 +1,13 @@
+%E-YAML
+---
+
+Name: XRandR
+Identifier: XRandR
+Author: Robert Colton
+Description: Adds Xlib display functions using the X Resize and Rotate extension.
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey]: Please add a note about how changing my display settings is an act of aggression, and if you attempt to do this, *they will  n e v e r   f i n d   y o u r   b o d y.*

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey]: Ok: 1c47c0f14eab3dcb94ede548d4325698e4802f00

Comment (JoshDreamland) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey]: You misspelled "never find your body."

Comment (JoshDreamland) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey]: Fixed that for you. e2bf020fb5ffce9d96f9fe1e68b94efb241c15c8

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Universal_System/Extensions/XRandR/About.ey]: Whatever, fair enough.

**Pull request merged**